api/models/build.js

```diff
@@ -44,25 +44,22 @@ const beforeValidate = (build) => {
 
 const sanitizeCompleteJobErrorMessage = message => message.replace(/\/\/(.*)@github/g, '//[token_redacted]@github');
 
-const completeJobErrorMessage = (err) => {
-  let message = 'An unknown error occurred';
-  if (err) {
-    message = err.message || err;
-  }
-  return sanitizeCompleteJobErrorMessage(message);
-};
+const jobErrorMessage = ({ message = 'An unknown error occurred' }) => sanitizeCompleteJobErrorMessage(message);
 
-const completeJobStateUpdate = (err, build, completedAt) => {
-  if (err) {
+const jobStateUpdate = (buildStatus, build, completedAt) => {
```

Comment:
If the garden build returns the expected status ("success" instead of "complete") then this logic can be simplified a bit.